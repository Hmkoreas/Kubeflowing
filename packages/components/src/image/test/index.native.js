/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { Image } from '../';

describe( 'Image', () => {
	it( 'renders nothing when src omitted', () => {
		const image = shallow( <Image /> );
		expect( image.type() ).toBeNull();
	} );

	it( 'renders an img tag with src and alt', () => {
		const image = shallow( <Image src="https://s.w.org/style/images/about/WordPress-logotype-wmark.png" alt="WordPress logo" /> );
		expect( image ).toMatchSnapshot();
	} );

	it( 'renders an img tag passing any additional props', () => {
		const image = shallow( <Image src="https://s.w.org/style/images/about/WordPress-logotype-wmark.png" alt="WordPress logo" data-id="123" /> );
		expect( image ).toMatchSnapshot();
	} );
} );
