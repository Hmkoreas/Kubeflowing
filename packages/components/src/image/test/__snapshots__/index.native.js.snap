// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders an img tag passing any additional props 1`] = `
<View
  style={
    Array [
      Object {
        "aspectRatio": undefined,
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Image
    accessibilityLabel="WordPress logo"
    data-id="123"
    source={
      Object {
        "uri": "https://s.w.org/style/images/about/WordPress-logotype-wmark.png",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`Image renders an img tag with src and alt 1`] = `
<View
  style={
    Array [
      Object {
        "aspectRatio": undefined,
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Image
    accessibilityLabel="WordPress logo"
    source={
      Object {
        "uri": "https://s.w.org/style/images/about/WordPress-logotype-wmark.png",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
